cpu/cc3200/periph/gpio.c

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (C) 2019 Ludwig Maximilian Universität
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup        cpu_cc3200
+ * @{
+ *
+ * @file
+ * @brief           GPIO driver implementation
+ *
+ * @author          Wladislaw Meixner <wladislaw.meixner@campus.lmu.de>
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "cpu.h"
+#include "periph/gpio.h"
+#include "vendor/hw_gpio.h"
+#include "vendor/hw_memmap.h"
+#include "vendor/hw_ocp_shared.h"
+#include "vendor/rom.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define GPIO_PINS_PER_PORT 8     /**< Number of pins per port */
+#define GPIO_DIR_MASK 0x00000001 /**< GPIO direction configuration mask */
+
+/**
+ * GPIO Pin type value used to configure pin to GPIO
+ */
+#define PIN_MODE_GPIO 0x00000000
+
+#define PAD_MODE_MASK 0x0000000F /**< GPIO mask for setting pin mode */
+/* GPIO mask for setting pin strength */
+#define PAD_STRENGTH_MASK 0x000000E0
+/* GPIO mask for setting pin type e.g. PULL */
+#define PAD_TYPE_MASK 0x00000310
+#define PAD_CONFIG_BASE ((OCP_SHARED_BASE + OCP_SHARED_O_GPIO_PAD_CONFIG_0))
+
+/* Define interrupt signal for GPIOA4 since it is not exported by TIs hw_ints.h
+ */
+#define INT_GPIOA4 20
+
+/**
+ * @brief get hardware configuration register for a pin
+ *
+ */
+#define PAD_CONFIG_REG(pin) \
+    (*((volatile uint32_t *)((gpio_pin << 2) + PAD_CONFIG_BASE)))
+
+/**
+ * @brief gpio base addresses
+ *
+ */
+static uint32_t ports[] = {
+    GPIOA0_BASE, GPIOA1_BASE, GPIOA2_BASE, GPIOA3_BASE, GPIOA4_BASE,
+};
+
+#ifdef MODULE_PERIPH_GPIO_IRQ
+/**
+ * @brief   static callback memory
+ */
+static gpio_isr_ctx_t isr_ctx[ARRAY_SIZE(ports)][GPIO_PINS_PER_PORT];
+#endif /* MODULE_PERIPH_GPIO_IRQ */
+
+/**
+ * @brief pin to GPIO pin numbers mappings
+ *
+ */
+static const uint8_t pin_to_gpio_num[64] = {
+    10,  11,  12,  13,  14,  15,  16,  17,  255, 255, 18,  19,  20,
+    21,  22,  23,  24,  40,  28,  29,  25,  255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 31,  255, 255, 255, 255, 0,   255, 32,
+    30,  255, 1,   255, 2,   3,   4,   5,   6,   7,   8,   9
+};
+
+/**
+ * @brief check if a port base is valid
+ *
+ * @param port
+ * @return true
+ * @return false
+ */
+bool _gpioPortBaseValid(uint32_t port)
```

Comment:
Is this used anywhere?
(if so it should probably be `static`)

Reply:
I was using this to validate port address before using it previously. But I dropped such checks since I did not see any other board do something linke that. So no it is not used anymore and I will remove it